modules/apps/transfer/keeper/msg_server_test.go

```diff
@@ -71,15 +71,53 @@ func (suite *KeeperTestSuite) TestMsgTransfer() {
 
 			tc.malleate()
 
-			res, err := suite.chainA.GetSimApp().TransferKeeper.Transfer(sdk.WrapSDKContext(suite.chainA.GetContext()), msg)
+			ctx := suite.chainA.GetContext()
+			res, err := suite.chainA.GetSimApp().TransferKeeper.Transfer(sdk.WrapSDKContext(ctx), msg)
+
+			expEvent := map[string]string{
+				"sender":            suite.chainA.SenderAccount.GetAddress().String(),
+				"receiver":          suite.chainB.SenderAccount.GetAddress().String(),
+				"amount":            coin.Amount.String(),
+				"denom":             coin.Denom,
+				"src_port":          path.EndpointA.ChannelConfig.PortID,
+				"src_channel":       path.EndpointA.ChannelID,
+				"dst_port":          path.EndpointB.ChannelConfig.PortID,
+				"dst_channel":       path.EndpointB.ChannelID,
+				"timeout_height":    suite.chainB.GetTimeoutHeight().String(),
+				"timeout_timestamp": "0",
+				"memo":              "memo",
+			}
+
+			checkTransferEvent := func(event sdk.Event) {
+				suite.Require().Equal(event.Type, types.EventTypeTransfer)
+				suite.Require().Len(event.Attributes, len(expEvent))
+				for _, attr := range event.Attributes {
+					expValue, found := expEvent[string(attr.Key)]
+					suite.Require().True(found)
+					suite.Require().Equal(expValue, string(attr.Value))
+				}
+			}
 
 			if tc.expPass {
 				suite.Require().NoError(err)
 				suite.Require().NotNil(res)
 				suite.Require().NotEqual(res.Sequence, uint64(0))
+
+				events := ctx.EventManager().Events()
+				var hasEvent bool
+				for _, event := range events {
```

Comment:
I am not sure if it would make more sense to move test for events to a seperate test case itself.
We could still have a very similar test as `TestMsgTransfer`, but check for events specially.

But i am ok with this structure as well.